src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1319,12 +1326,24 @@ public static void writeAdditionalLevelSaveData(WorldData worldData, CompoundTag
         levelTag.put("fml", fmlData);
     }
 
-    public static void readAdditionalLevelSaveData(CompoundTag rootTag)
+    /**
+     * @deprecated To be removed in 1.20.
+     * Use {@link #readAdditionalLevelSaveData(LevelStorageSource.LevelDirectory, CompoundTag)} instead.
+     */
+    @Deprecated(forRemoval = true, since = "1.19.2")
+    public static void readAdditionalLevelSaveData(CompoundTag rootTag) {
+        readAdditionalLevelSaveData(null, rootTag);
+    }
+
+    @ApiStatus.Internal
+    public static void readAdditionalLevelSaveData(LevelStorageSource.LevelDirectory levelDirectory, CompoundTag rootTag)
```

Comment:
This should have the level directory marked as Nullable, along with cascading that annotation to the event and any getters. Why exactly are you marking this new method as an internal api?